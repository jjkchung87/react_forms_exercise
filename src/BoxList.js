import React, {useState} from 'react'
import Box from './Box'
import NewBoxForm from './NewBoxForm'
import {v4 as uuid} from 'uuid'


const BoxList = () => {

    const [boxes, setBoxes] = useState([{color:'red', height: '20px', width: '20px', id: uuid()}])

    const addBox = ({width, height, color}) => {
        setBoxes(boxes => ([...boxes,{width, height, color, id: uuid()}]))
    }

    const deleteBox = (deletedBox) => {
        setBoxes(boxes => (boxes.filter(box => box.id !== deletedBox.id)))
    }

    return (
        <div className="BoxList">
            <NewBoxForm addBox={addBox} />
            {boxes.map(box => <Box 
                                color={box.color} 
                                height={box.height} 
                                width={box.width} 
                                id={box.id} 
                                key={box.id} 
                                deleteBox={() => deleteBox(box)}
                                />)} 
        </div>
        
    )

}

export default BoxList;