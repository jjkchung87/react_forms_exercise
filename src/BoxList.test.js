import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

describe('<BoxList /> rendering', function () {
    describe('inital BoxList', function () {
        it('renders without crashing', function () {
        render(<BoxList />);
        });

        it('matches snapshot', function () {
        const { asFragment } = render(<BoxList />);
        expect(asFragment()).toMatchSnapshot();
        });

    });
});

describe('delete box', function () {
    it('delete box', function () {
        const boxList = render(<BoxList />);

        const btn = boxList.queryByText('x');
      
        // Assert that the box is initially in the document
        // const box = boxList.querySelector('.Box');
      
        // Click on the delete button
        fireEvent.click(btn);

        // box to not be on List
        expect(btn).not.toBeInTheDocument();

    });
});


describe('create box', function () {
    it('create box', function () {
        const { getByLabelText,getByPlaceholderText, queryByText, container } = render(<BoxList />);
        
        const heightInput = getByPlaceholderText('height')
        const widthInput = getByPlaceholderText('width')
        const colorInput = getByPlaceholderText('color')
        const button = queryByText('Submit')

        fireEvent.change(heightInput, {target: {value: '50px'}})
        fireEvent.change(widthInput, {target: {value: '50px'}})
        fireEvent.change(colorInput, {target: {value: 'green'}})

        fireEvent.click(button)

        expect(container.querySelectorAll('.Box')).toHaveLength(2)


    });
});


