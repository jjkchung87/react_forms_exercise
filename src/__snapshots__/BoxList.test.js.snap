// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxList /> rendering inital BoxList and win state matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        value=""
      />
      <label
        for="width"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        value=""
      />
      <label
        for="width"
      >
        Color: 
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div
      class="Box"
      style="background-color: red; height: 20px; width: 20px;"
    />
    <span
      class="Box-delete"
    >
      x
    </span>
  </div>
</DocumentFragment>
`;

exports[`<BoxList /> rendering inital BoxList and win state matches snapshot BoxList 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        value=""
      />
      <label
        for="width"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        value=""
      />
      <label
        for="width"
      >
        Color: 
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div
      class="Box"
      style="background-color: red; height: 20px; width: 20px;"
    />
    <span
      class="Box-delete"
    >
      x
    </span>
  </div>
</DocumentFragment>
`;

exports[`<BoxList /> rendering inital BoxList and win state matches snapshot for full BoxList 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        value=""
      />
      <label
        for="width"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        value=""
      />
      <label
        for="width"
      >
        Color: 
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div
      class="Box"
      style="background-color: red; height: 20px; width: 20px;"
    />
    <span
      class="Box-delete"
    >
      x
    </span>
  </div>
</DocumentFragment>
`;

exports[`<BoxList /> rendering inital BoxList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        value=""
      />
      <label
        for="color"
      >
        Color: 
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div
      class="Box"
      style="background-color: red; height: 20px; width: 20px;"
    />
    <span
      class="Box-delete"
    >
      x
    </span>
  </div>
</DocumentFragment>
`;
